src/adapters/CKEditorAdapter.ts

```diff
@@ -68,6 +97,10 @@ export class CKEditorAdapter extends AbstractRichtextEditorAdapter {
   }
 
   isInWysiwygMode() {
-    return this.getEditor().mode === 'wysiwyg';
+    if(this.editorVersion == 4){
+      return this.getEditor().mode === 'wysiwyg';
+    }
+    const editorElement = this.getEditorDocument().querySelector('.ck-editor .ck-source-editing-button')!;
```

Comment:
Please avoid using css selectors on buttons where possible, as there is high chance that they may break in minor updates also they might get overridden by the CMS application where CK editor is embedded.

Try using CK5 API

in CK5 source view is available as native plugin.
You can use the API to check if its enabled and currently in use.

Source Editing Demo: https://ckeditor.com/docs/ckeditor5/latest/features/source-editing.html

Source Editing API: https://ckeditor.com/docs/ckeditor5/latest/api/module_source-editing_sourceediting-SourceEditing.html


```
if (instance.plugins.has("SourceEditing")) {
  const sep = instance.plugins.get("SourceEditing");
  return sep.isEnabled && sep.isSourceEditingMode;
}
```

The plugin id "SourceEditing"is taken from docs: https://ckeditor.com/docs/ckeditor5/latest/api/module_source-editing_sourceediting-SourceEditing.html